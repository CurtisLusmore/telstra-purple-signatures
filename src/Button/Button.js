import React from 'react';
import { compose, withState, withHandlers, mapProps } from 'recompose';
import { pick } from 'lodash/object';

const Button = ({ classState, textState, onClick }) => (
  <button type="button" className={classState} onClick={onClick}>
    {textState}
  </button>
);

export default compose(
  withState('classState', 'setClass', ({ classBefore }) => classBefore),
  withState('textState', 'setText', ({ textBefore }) => textBefore),
  withHandlers({
    onClick: ({
      setClass,
      setText,
      onClickHandler,
      classAfter,
      textAfter,
      classBefore,
      textBefore
    }) => event => {
      onClickHandler(event);
      setClass(classAfter, () => setTimeout(() => setClass(classBefore), 1000));
      setText(textAfter, () => setTimeout(() => setText(textBefore), 1000));
    }
  }),
  mapProps(props => pick(props, ['classState', 'textState', 'onClick']))
)(Button);
