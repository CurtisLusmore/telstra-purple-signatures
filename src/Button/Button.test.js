import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

it('Renders', () => {
  const tree = shallow(
    <Button
      onClickHandler={() => null}
      classBefore={'test before'}
      classAfter={'test after'}
      textBefore={'text before'}
      textAfter={'text after'}
    />
  );
  expect(tree).toMatchSnapshot();
});
