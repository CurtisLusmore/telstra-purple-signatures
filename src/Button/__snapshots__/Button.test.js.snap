// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<withState(withHandlers(mapProps(Button)))
  classAfter="test after"
  classBefore="test before"
  classState="test before"
  onClickHandler={[Function]}
  setClass={[Function]}
  textAfter="text after"
  textBefore="text before"
/>
`;
